tests/components/blue_current/test_init.py

```diff
@@ -103,3 +107,85 @@ async def test_connect_request_limit_reached_error(
     await started_loop.wait()
     assert mock_client.get_next_reset_delta.call_count == 1
     assert mock_client.connect.call_count == 2
+
+
+async def test_start_charging_action(
+    hass: HomeAssistant, config_entry: MockConfigEntry
+) -> None:
+    """Test the start charing action when a charging card is provided."""
+    integration = await init_integration(hass, config_entry, Platform.BUTTON)
+    client = integration[0]
+
+    await hass.services.async_call(
+        DOMAIN,
+        "start_charge_session",
+        {
+            "device_id": list(dr.async_get(hass).devices)[0],
+            "charging_card_id": "TEST_CARD",
+        },
+        blocking=True,
+    )
+
+    client.start_session.assert_called_once_with("101", "TEST_CARD")
+
+
+async def test_start_charging_action_without_card(
+    hass: HomeAssistant, config_entry: MockConfigEntry
+) -> None:
+    """Test the start charing action when no charging card is provided."""
+    integration = await init_integration(hass, config_entry, Platform.BUTTON)
+    client = integration[0]
+
+    await hass.services.async_call(
+        DOMAIN,
+        "start_charge_session",
+        {
+            "device_id": list(dr.async_get(hass).devices)[0],
+        },
+        blocking=True,
+    )
+
+    client.start_session.assert_called_once_with("101", "BCU-APP")
+
+
+async def test_start_charging_action_errors(
+    hass: HomeAssistant, config_entry: MockConfigEntry
+) -> None:
+    """Test the start charing action errors."""
+    await init_integration(hass, config_entry, Platform.BUTTON)
+
+    with pytest.raises(ServiceValidationError):
+        # No device id
+        await hass.services.async_call(
+            DOMAIN,
+            "start_charge_session",
+            {},
+            blocking=True,
+        )
+
+    with pytest.raises(ServiceValidationError):
+        # Invalid device id
+        await hass.services.async_call(
+            DOMAIN,
+            "start_charge_session",
+            {"device_id": "INVALID"},
+            blocking=True,
+        )
+
+    get_entry_mock = MagicMock()
+    get_entry_mock.state = ConfigEntryState.SETUP_IN_PROGRESS
+
+    with (
+        patch.object(
+            hass.config_entries, "async_get_entry", return_value=get_entry_mock
+        ),
+        pytest.raises(ServiceValidationError),
+    ):
```

Comment:
I think it's practically impossible to have no config entry when you use the config entry id from a device, so I would just test that a mock config entry is not loaded